src/ConfigParser.cc

```diff
@@ -378,9 +378,7 @@ ConfigParser::NextToken()
 
             ConfigParser::CfgFile *wordfile = new ConfigParser::CfgFile();
             if (!path || !wordfile->startParse(path)) {
-                debugs(3, DBG_CRITICAL, "FATAL: Error opening config file: " << token);
                 delete wordfile;
-                self_destruct();
                 return nullptr;
```

Comment:
> We need to handle both cases similarly

I agree. However, it would be nice to find a way to handle them similarly _without_ relaxing official default parameters() handling because parameters() handling default (a fatal error) is better than the current "quoted file" default (an error message).

> let the ConfigParser caller decide whether an empty or missing configuration is fatal or not.

Perhaps we can delay this PR changes until ConfigParser caller defaults become as conservative as the current official parameters() code? At that time, we can unify the two parameter handling paths without relaxing the more conservative one.

P.S. My concern above ignores the secondary fact that self_destruct() should be (eventually) replaced with an exception.